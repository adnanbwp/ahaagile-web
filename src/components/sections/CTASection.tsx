'use client';

import React from 'react';
import { Button } from '@/components/ui/button';
import { ArrowRight } from '@/lib/icons';

interface CTASectionProps {}

export default function CTASection(): JSX.Element {
  const handleScrollToCalendly = () => {
    const calendlyElement = document.getElementById('calendly-widget');
    if (calendlyElement) {
      calendlyElement.scrollIntoView({ behavior: 'smooth' });
    }
  };

  return (
    <section 
      className="relative py-20 px-4 bg-gradient-to-br from-primary-800 via-primary-700 to-primary-900 overflow-hidden"
      aria-labelledby="cta-heading"
      role="region"
    >
      {/* Background gradient overlay - using design system colors properly */}
      <div className="absolute inset-0 bg-gradient-to-br from-primary-800/95 via-primary-700/95 to-primary-900/95"></div>
      
      {/* Content container */}
      <div className="container mx-auto max-w-6xl relative z-10">
        <div className="text-center max-w-4xl mx-auto">
          {/* Main headline */}
          <h2 
            id="cta-heading"
            className="text-4xl md:text-5xl lg:text-6xl font-bold text-white mb-8 font-heading leading-tight"
          >
            Ready to Get Started?
          </h2>
          
          {/* Reassuring copy */}
          <div className="text-xl md:text-2xl text-primary-100 mb-12 font-serif leading-relaxed space-y-6">
            <p>
              Book your free consultation today and discover how intelligent workflow automation can transform your business operations.
            </p>
            <p className="text-lg md:text-xl text-primary-200">
              Our expert team will analyze your current processes and provide actionable recommendations tailored to your specific needs.
            </p>
            <p className="text-base md:text-lg text-primary-300">
              No commitment required - just valuable insights to help you make informed decisions about your workflow optimization journey.
            </p>
          </div>
          
          {/* CTA Buttons */}
          <div className="flex flex-col sm:flex-row gap-6 justify-center items-center">
            {/* Primary CTA - Book Consultation Now */}
            <Button 
              onClick={handleScrollToCalendly}
              size="lg"
              className="bg-accent-400 hover:bg-accent-500 text-primary-900 font-semibold px-8 py-4 h-auto text-lg font-heading shadow-xl hover:shadow-2xl transform hover:scale-105 hover:-translate-y-1 transition-all duration-300 group focus:ring-2 focus:ring-accent-300 focus:outline-none"
              aria-label="Book your free consultation - scroll to scheduling widget"
            >
              Book Consultation Now
              <ArrowRight 
                className="ml-2 h-5 w-5 group-hover:translate-x-1 transition-transform duration-300" 
                aria-hidden="true"
              />
            </Button>
            
            {/* Secondary CTA - Have Questions? Contact Us */}
            <Button 
              asChild
              variant="outline"
              size="lg"
              className="border-2 border-primary-100 text-primary-100 hover:bg-primary-100 hover:text-primary-900 font-semibold px-8 py-4 h-auto text-lg font-heading shadow-lg hover:shadow-xl transform hover:scale-105 hover:-translate-y-1 transition-all duration-300 bg-transparent focus:ring-2 focus:ring-primary-200 focus:outline-none"
            >
              <a 
                href="mailto:info@ahaagile.com"
                aria-label="Send email to info@ahaagile.com for questions about our services"
              >
                Have Questions? Contact Us
              </a>
            </Button>
          </div>
        </div>
      </div>
      
      {/* Decorative elements */}
      <div className="absolute top-0 right-0 w-64 h-64 bg-accent-400 opacity-10 rounded-full blur-3xl transform translate-x-32 -translate-y-32"></div>
      <div className="absolute bottom-0 left-0 w-48 h-48 bg-accent-400 opacity-10 rounded-full blur-3xl transform -translate-x-24 translate-y-24"></div>
    </section>
  );
}