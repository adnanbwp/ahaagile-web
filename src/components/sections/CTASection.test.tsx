import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import CTASection from './CTASection';

// Mock scrollIntoView since it's not available in JSDOM
const mockScrollIntoView = jest.fn();
Object.defineProperty(HTMLElement.prototype, 'scrollIntoView', {
  configurable: true,
  value: mockScrollIntoView,
});

describe('CTASection', () => {
  beforeEach(() => {
    mockScrollIntoView.mockClear();
  });

  it('renders with correct headline and copy', () => {
    render(<CTASection />);
    
    // Test main headline
    expect(screen.getByText('Ready to Get Started?')).toBeInTheDocument();
    
    // Test reassuring copy content
    expect(screen.getByText(/Book your free consultation today/)).toBeInTheDocument();
    expect(screen.getByText(/Our expert team will analyze/)).toBeInTheDocument();
    expect(screen.getByText(/No commitment required/)).toBeInTheDocument();
  });

  it('renders primary CTA button with correct text and arrow icon', () => {
    render(<CTASection />);
    
    const primaryButton = screen.getByRole('button', { name: /Book your free consultation - scroll to scheduling widget/ });
    expect(primaryButton).toBeInTheDocument();
    expect(primaryButton).toHaveTextContent('Book Consultation Now');
    
    // Check for arrow icon (Lucide React ArrowRight)
    const arrowIcon = primaryButton.querySelector('svg');
    expect(arrowIcon).toBeInTheDocument();
    expect(arrowIcon).toHaveAttribute('aria-hidden', 'true');
  });

  it('renders secondary CTA button with correct text and outline styling', () => {
    render(<CTASection />);
    
    const secondaryButton = screen.getByRole('link', { name: /Send email to info@ahaagile.com/ });
    expect(secondaryButton).toBeInTheDocument();
    expect(secondaryButton).toHaveTextContent('Have Questions? Contact Us');
    expect(secondaryButton).toHaveAttribute('href', 'mailto:info@ahaagile.com');
  });

  it('handles primary button click to scroll to Calendly widget', () => {
    // Mock getElementById to return an element
    const mockCalendlyElement = document.createElement('div');
    mockCalendlyElement.id = 'calendly-widget';
    jest.spyOn(document, 'getElementById').mockReturnValue(mockCalendlyElement);
    
    render(<CTASection />);
    
    const primaryButton = screen.getByRole('button', { name: /Book your free consultation - scroll to scheduling widget/ });
    fireEvent.click(primaryButton);
    
    expect(mockScrollIntoView).toHaveBeenCalledWith({ behavior: 'smooth' });
  });

  it('handles primary button click when Calendly widget not found', () => {
    // Mock getElementById to return null
    jest.spyOn(document, 'getElementById').mockReturnValue(null);
    
    render(<CTASection />);
    
    const primaryButton = screen.getByRole('button', { name: /Book your free consultation - scroll to scheduling widget/ });
    fireEvent.click(primaryButton);
    
    // Should not throw error and should not call scrollIntoView
    expect(mockScrollIntoView).not.toHaveBeenCalled();
  });

  it('has proper semantic structure and accessibility features', () => {
    render(<CTASection />);
    
    // Test section element with proper ARIA attributes
    const section = screen.getByRole('region');
    expect(section).toHaveAttribute('aria-labelledby', 'cta-heading');
    
    // Test heading with proper ID
    const heading = screen.getByText('Ready to Get Started?');
    expect(heading).toHaveAttribute('id', 'cta-heading');
    expect(heading.tagName).toBe('H2');
    
    // Test button accessibility
    const primaryButton = screen.getByRole('button', { name: /Book your free consultation - scroll to scheduling widget/ });
    expect(primaryButton).toHaveAttribute('aria-label');
    
    const secondaryButton = screen.getByRole('link', { name: /Send email to info@ahaagile.com/ });
    expect(secondaryButton).toHaveAttribute('aria-label');
  });

  it('applies correct brand colors and styling classes', () => {
    render(<CTASection />);
    
    // Test section has gradient background
    const section = screen.getByRole('region');
    expect(section).toHaveClass('bg-gradient-to-br', 'from-primary-800', 'via-primary-700', 'to-primary-900');
    
    // Test primary button styling
    const primaryButton = screen.getByRole('button', { name: /Book your free consultation - scroll to scheduling widget/ });
    expect(primaryButton).toHaveClass('bg-accent-400', 'hover:bg-accent-500', 'text-primary-900');
    
    // Test secondary button styling
    const secondaryButton = screen.getByRole('link', { name: /Send email to info@ahaagile.com/ });
    expect(secondaryButton).toHaveClass('border-primary-100', 'text-primary-100');
  });

  it('includes hover animation classes for interactive elements', () => {
    render(<CTASection />);
    
    // Test primary button hover animations
    const primaryButton = screen.getByRole('button', { name: /Book your free consultation - scroll to scheduling widget/ });
    expect(primaryButton).toHaveClass('hover:scale-105', 'hover:-translate-y-1', 'transition-all', 'duration-300');
    
    // Test secondary button hover animations
    const secondaryButton = screen.getByRole('link', { name: /Send email to info@ahaagile.com/ });
    expect(secondaryButton).toHaveClass('hover:scale-105', 'hover:-translate-y-1', 'transition-all', 'duration-300');
    
    // Test arrow icon hover animation
    const arrowIcon = primaryButton.querySelector('svg');
    expect(arrowIcon).toHaveClass('group-hover:translate-x-1', 'transition-transform', 'duration-300');
  });

  it('uses correct typography classes for brand consistency', () => {
    render(<CTASection />);
    
    // Test heading uses Inter font family
    const heading = screen.getByText('Ready to Get Started?');
    expect(heading).toHaveClass('font-heading');
    
    // Test body text uses Source Serif 4 font family
    const bodyText = screen.getByText(/Book your free consultation today/);
    expect(bodyText.closest('div')).toHaveClass('font-serif');
  });

  it('has proper responsive layout classes', () => {
    render(<CTASection />);
    
    // Test button container responsive layout
    const buttonContainer = screen.getByRole('button', { name: /Book your free consultation - scroll to scheduling widget/ }).parentElement;
    expect(buttonContainer).toHaveClass('flex', 'flex-col', 'sm:flex-row', 'gap-6');
    
    // Test responsive text sizing
    const heading = screen.getByText('Ready to Get Started?');
    expect(heading).toHaveClass('text-4xl', 'md:text-5xl', 'lg:text-6xl');
  });

  it('includes decorative elements with proper positioning', () => {
    const { container } = render(<CTASection />);
    
    // Test decorative gradient circles
    const decorativeElements = container.querySelectorAll('.bg-accent-400.opacity-10.rounded-full.blur-3xl');
    expect(decorativeElements).toHaveLength(2);
    
    // Test positioning classes
    expect(decorativeElements[0]).toHaveClass('absolute', 'top-0', 'right-0');
    expect(decorativeElements[1]).toHaveClass('absolute', 'bottom-0', 'left-0');
  });

  it('matches snapshot for visual regression testing', () => {
    const { container } = render(<CTASection />);
    expect(container.firstChild).toMatchSnapshot();
  });
});