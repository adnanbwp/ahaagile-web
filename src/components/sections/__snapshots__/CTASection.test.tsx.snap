// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTASection matches snapshot for visual regression testing 1`] = `
<section
  aria-labelledby="cta-heading"
  class="relative py-20 px-4 bg-gradient-to-br from-primary-800 via-primary-700 to-primary-900 overflow-hidden"
  role="region"
>
  <div
    class="absolute inset-0 bg-gradient-to-br from-primary-800/95 via-primary-700/95 to-primary-900/95"
  />
  <div
    class="container mx-auto max-w-6xl relative z-10"
  >
    <div
      class="text-center max-w-4xl mx-auto"
    >
      <h2
        class="text-4xl md:text-5xl lg:text-6xl font-bold text-white mb-8 font-heading leading-tight"
        id="cta-heading"
      >
        Ready to Get Started?
      </h2>
      <div
        class="text-xl md:text-2xl text-primary-100 mb-12 font-serif leading-relaxed space-y-6"
      >
        <p>
          Book your free consultation today and discover how intelligent workflow automation can transform your business operations.
        </p>
        <p
          class="text-lg md:text-xl text-primary-200"
        >
          Our expert team will analyze your current processes and provide actionable recommendations tailored to your specific needs.
        </p>
        <p
          class="text-base md:text-lg text-primary-300"
        >
          No commitment required - just valuable insights to help you make informed decisions about your workflow optimization journey.
        </p>
      </div>
      <div
        class="flex flex-col sm:flex-row gap-6 justify-center items-center"
      >
        <button
          aria-label="Book your free consultation - scroll to scheduling widget"
          class="inline-flex items-center justify-center whitespace-nowrap ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 rounded-md bg-accent-400 hover:bg-accent-500 text-primary-900 font-semibold px-8 py-4 h-auto text-lg font-heading shadow-xl hover:shadow-2xl transform hover:scale-105 hover:-translate-y-1 transition-all duration-300 group focus:ring-2 focus:ring-accent-300 focus:outline-none"
        >
          Book Consultation Now
          <svg
            aria-hidden="true"
            class="lucide lucide-arrow-right ml-2 h-5 w-5 group-hover:translate-x-1 transition-transform duration-300"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 12h14"
            />
            <path
              d="m12 5 7 7-7 7"
            />
          </svg>
        </button>
        <a
          aria-label="Send email to info@ahaagile.com for questions about our services"
          class="inline-flex items-center justify-center whitespace-nowrap ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 rounded-md border-2 border-primary-100 text-primary-100 hover:bg-primary-100 hover:text-primary-900 font-semibold px-8 py-4 h-auto text-lg font-heading shadow-lg hover:shadow-xl transform hover:scale-105 hover:-translate-y-1 transition-all duration-300 bg-transparent focus:ring-2 focus:ring-primary-200 focus:outline-none"
          href="mailto:info@ahaagile.com"
        >
          Have Questions? Contact Us
        </a>
      </div>
    </div>
  </div>
  <div
    class="absolute top-0 right-0 w-64 h-64 bg-accent-400 opacity-10 rounded-full blur-3xl transform translate-x-32 -translate-y-32"
  />
  <div
    class="absolute bottom-0 left-0 w-48 h-48 bg-accent-400 opacity-10 rounded-full blur-3xl transform -translate-x-24 translate-y-24"
  />
</section>
`;
